generators/entity-server/templates/partials/save_template.ejs

```diff
@@ -46,18 +46,18 @@ if (isUsingMapsId) {
         <%= mapsIdAssoc.otherEntity.primaryKey.type %> <%= otherEntityName %>Id = <%= instanceName %>.get<%= mapsIdAssoc.relationshipNameCapitalized %>().get<%= primaryKey.nameCapitalized %>();
         <%= mapsIdRepoInstance %>.findById(<%= otherEntityName %>Id).ifPresent(<%= persistInstance %>::<%_ if (!fluentMethods) { _%>set<%= mapsIdAssoc.relationshipNameCapitalized %> <%_ } else { _%><%= mapsIdAssoc.relationshipName %><%_ } _%>);
     <%_ } _%>
-        <%= persistInstance %> = <%= entityInstance %>Repository.save(<%= persistInstance %>);
```

Comment:
Save is really the keyword here: what about the rollback if there's something that fails afterwards?
We are loosing all the benefits of the Transactions here.

Reply:
Save and flush does not commit so transaction can still be rolled back.  And with just save  this data is not picked up while indexing later.  ( more about rationale in the other comment )

Reply:
Always not a fan of it: stacking database queries is way better in term of perf.
Is it really mandatory?

Reply:
I like to ensure that indexing really gets complete updated database state -  because entity object here may not have all the relations initialized properly ( it was converted from DTO ).   That's why  I do indexing asynchronously,   and in separate transaction ( See EntitySearchRepository.java.ejs @ 164-171 ).   Save and flush ensures that READ_COMMITTED isolation level  kicks in even in asynchronous scenario.

Reply:
@ko5tik Sure :-).
The Thing that I'm not sure about is if it makes sense to persist the related elements in the ES database as well.
Let me try to elaborate:
- When you query an element, you basically just want that element and not necessarly its related ones.
- If you really want the siblings you could always 'findByIdWithEagerRelationship' from the database.

I'm not saying I'm against, both seems valid; The question being: do we want to have the exact same behaviour for search than the one we have with sql

Reply:
Standard behaviour of elasticsearch would be to persist complete object tree (can be modified by annotations when necessary).    I my use case I need to be able to find entities by dependent objects ( say, customer conversations based on phone calls time of day )  -  thus  I need them indexed together.  And when  conversation object is updated and reindexed,  I will not lose related objects  from index - which is the actual case.

PS:  As this branch was too old and synch  brought in too many commits  I recreated it and raised a new pull request

https://github.com/jhipster/generator-jhipster/pull/17780